File: lib/core/util/currency_formatter.dart
import 'package:intl/intl.dart';

class CurrencyFormatter {
  static String format(double amount, [String currency = "SAR"]) {
    final formatter = NumberFormat("#,##0.00", "en_US");
    return "${formatter.format(amount)} $currency";
  }

  static String formatCompact(double amount, [String currency = "SAR"]) {
    if (amount >= 1000000) {
      final val = (amount / 1000000).toStringAsFixed(2);
      return "$val M $currency";
    } else if (amount >= 10000) {
      final val = (amount / 1000).toStringAsFixed(1);
      return "$val K $currency";
    } else {
      return format(amount, currency);
    }
  }
}

File: app/src/main/java/com/example/core/util/CurrencyFormatter.kt
package com.example.core.util

import java.text.NumberFormat
import java.util.Locale

object CurrencyFormatter {
    fun format(amount: Double, currency: String = "SAR"): String {
        val formatter = NumberFormat.getCurrencyInstance(Locale.US)
        val formattedNumber = String.format(Locale.US, "%,.2f", amount)
        return "$formattedNumber $currency"
    }

    fun formatCompact(amount: Double, currency: String = "SAR"): String {
        return when {
            amount >= 1_000_000 -> String.format(Locale.US, "%.2fM %s", amount / 1_000_000, currency)
            amount >= 10_000 -> String.format(Locale.US, "%.1fK %s", amount / 1_000, currency)
            else -> format(amount, currency)
        }
    }
}
